### ISSUE TITLE
Dashboard support in Kueue

### ISSUE THREAD

**Issue by kerthcet** [kubernetes-sigs/kueue#940]
**What would you like to be added**:

It would be great if we can have an insight about what's our queueing system looks like at real time
- for administrators, it helps to understand the total resource usages in-between cluster queues and whether we should make them a cohort
- for batch users, they will have an overview about the job queueing, how many jobs are pending for scheduling, how long jobs are waiting.

Overall, it's a great enhancement especially for production env.

**Why is this needed**:

A big enhancement and a great insight of kueue system.

**Completion requirements**:

This enhancement requires the following artifacts:

- [x] Design doc
- [ ] API change
- [x] Docs update

The artifacts should be linked in subsequent comments.

Some advices here:
- For administrators:
  - clusterQueue resource groups
  - clusterQueue resource utilizations
  - localQueues
  - cohorts
- For users:
  - jobs with their status

**Comment by alculquicondor** [comment]
Do you have a high level idea of how to get there?

For example, would metrics + grafana yamls be enough for the administrative side?

For end users, certainly grafana wouldn't be viable. But what could be the MVP that would keep kueue largely non-opinionated and reusable (so you can integrate it with your own UI, if you already have one). Could we offer a CLI instead?

**Comment by moficodes** [comment]
Kueue already spits out prometheus metrics. Building a UI based on that can be useful and the UI should be optional to deploy. 

I do wonder if it is more useful for us to provide general purpose grafana dashboard and make it available in https://grafana.com/grafana/dashboards/

**Comment by kerthcet** [comment]
Building on metrics is helpful I think, but the dashboard is more than that, like it will display the basic information about the system, how many queues there, what their names are, how many jobs inside the queue, it can be interactive. We can get the information via the apis directly or we can have a lightweight database inside for cache, like sqlite. 

We may need some frontend volunteers if we want to finish this work. As a MVP, IMHO, I think it should include
- Most of the API objects(clusterQueue, localQueue, Job, resourceFlavor, workload) at least, also including their relationships
- Some exported metrics

**Comment by ahg-g** [comment]
+1000, this is a much needed experience gap, I would be happy to review proposals.

@kerthcet may be we start by looking at similar batch schedulers and see what "screens" they offer to inform and help seed what we need to build?

**Comment by zeddit** [comment]
@ahg-g 
it's a core feature which will make kueue easy to use.
A alternative would be airflow, and its UI looks like below
![image](https://github.com/kubernetes-sigs/kueue/assets/30164206/afc164a3-2b6f-4285-b000-ef1a197f07cf)
it contains task status, code, and audit logs which would be useful information for user to inspect their jobs.
besides, airflow integrates with idp like ldap or oidc and provides access control and permission managment features.

However, airflow doesn't provide an API to submit one-time run jobs like ml training jobs, which is the core application for kueue.

**Comment by kerthcet** [comment]
cc @B1F030 we also did some research around the popular queueing systems, I think we can provide a summary about the   essential elements in dashboard, or even a prototype. Can you help with this @B1F030 ?

**Comment by samzong** [comment]
Hi guys, I want to try involved in the prototyping part of the dashbaord Desgin, and provide the prototype like figma.

**Comment by kerthcet** [comment]
Thanks @samzong 
We can provide a based design, and share with the community for feedbacks and then involve the developments, any concerns? @alculquicondor

**Comment by alculquicondor** [comment]
Maybe we can start with a list of views you would like to have and do priority sorting

**Comment by kerthcet** [comment]
> Maybe we can start with a list of views you would like to have and do priority sorting

@B1F030 is doing this.

**Comment by kerthcet** [comment]
Thanks @Sharpz7 that's helpful, and we have a general idea now, @samzong is doing the prototyping, once we've done, we'll share a google doc/figma with your guys, hope to work together.

**Comment by Sharpz7** [comment]
Great, Thank you :))

**Comment by kerthcet** [comment]
Are you still following this @samzong ?

**Comment by samzong** [comment]
@kerthcet 

Absolutely, I'm still very interested in this. The good news is that I'll have more time to contribute to open source in the coming period. I'll make sure to push this forward as soon as possible.

**Comment by trashadewan** [comment]
Hey, 
there is a grafana dashboard shown in https://www.youtube.com/watch?v=B63vT2_UYE4, would you know if this is avialble for use. somewhere?

**Comment by alculquicondor** [comment]
I think it might be this one https://github.com/GoogleCloudPlatform/ai-on-gke/blob/main/best-practices/gke-batch-refarch/02_platform/monitoring/deploy-dashboard.yaml

but @moficodes and @alizaidis to confirm

**Comment by alizaidis** [comment]
Yep that's the one!

**Comment by akram** [comment]
Hi everyone,

As a side project to just run a demo I have contributed kueue-viz : as a kueue dashboard.
The project is availabe here: https://github.com/akram/kueue-viz

It is still very basic , but every contribution and feedback are welcome.


<img width="1511" alt="image" src="https://github.com/user-attachments/assets/5728e0cf-2aad-4c27-9511-78da1d995347">

**Comment by kannon92** [comment]
@mwielgus has mentioned that this is still of interest to the Kueue project.

**Comment by kerthcet** [comment]
Thanks @akram for the work!

**Comment by mimowo** [comment]
/reopen
let's use this issue to continue with the next steps of productazing the kueue-viz sub-project. I post some steps in the [comment](https://github.com/kubernetes-sigs/kueue/pull/3727#issuecomment-2519804009)

**Comment by mimowo** [comment]
/remove-lifecycle stale
This is being actively worked on

**Comment by mbobrovskyi** [comment]
@mimowo @akram Looks like we have the Dashboard now. Can we close this issue, or is there still something pending?

**Comment by mimowo** [comment]
For the functional requirements I will defer to @akram and @kannon92 the question. For the technical I would like to complete this https://github.com/kubernetes-sigs/kueue/issues/3796 too.

**Comment by kannon92** [comment]
Let's keep this open until 0.12 release. I think technically we don't yet have this as it is not released.

**Comment by shaharelys** [comment]
Hey guys! We would really like to try this one for our use case. I saw this is on the roadman for 2025 but I realize now this is already available? Is that correct?

Would love to be an alpha user. Could also contibute from the usecase POV of an administrator for a 50 H100s k8s cluster with 10 ClusterQueues :)

Is that possible? Thanks!

**Comment by kannon92** [comment]
I change my mind on this. We should close it. It will be released in 0.12.
